src/components/Admin/addAdmin.tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useState } from "react";
+import { Plus } from "lucide-react";
+import { Controller, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+import { axiosInstance } from "@/lib/axiosinstance";
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+  DialogTitle,
+  DialogDescription,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectTrigger,
+  SelectItem,
+  SelectValue,
+  SelectContent,
+} from "@/components/ui/select";
+
+interface AdminData {
+  firstName: string;
+  lastName: string;
+  userName: string;
+  email: string;
+  phone: string;
+  type: "Admin" | "Super_Admin";
+  status: "Pending"; // status is fixed to "Pending"
+}
+interface UserData {
+  _id: string;
+  firstName: string;
+  lastName: string;
+  userName: string;
+  email: string;
+  phone: string;
+  type: string;
+  status: string;
+  createdAt: string;
+  updatedAt: string;
+}
+interface AddAdminProps {
+  onAddDomain: (newDomain: UserData) => void;
+}
+const adminSchema = z.object({
```

Comment:
Create the enum file we discussed in meeting and  make an object containing all of following nonempty prompt message 
Eg:
In enums file
const adminSchemaPromptMessages = {
FIRST_NAME_REQUIRED: 'Please enter the first name',
}